Modify S1155: Migrate to LaYC (#3327)
SonarSource/rspec: rules/S1155/cfamily/rule.adoc

Reviewer: The wording suggestions of @frederic-tingaud-sonarsource look good, IMO; you could replicate them in the other `rule.adoc` as the descriptions are close.
About the deletion of the `How to fix it` part to integrate the code examples in the `Why is this an issue?` part, I personally like to keep both, even for "simple" rules, so I let you do as you prefer.

PS: an empty line can be removed at lines 52/53.

== Why is this an issue?

> When you call `empty()` or `is_empty()`, it clearly communicates the intention of the code, which is to check if the collection is empty. Using `size() == 0` for this purpose is less direct and can make the code slightly harder to understand.

Reviewer 2: ```suggestion
When you call `empty()` or `is_empty()`, it clearly communicates the intention of the code, which is to check if the collection is empty. Using `size() == 0` for this purpose is less direct and makes the code slightly more complex.
```

Reviewer: Minor: `the intention of the code` can be replaced by `the code's intention`

> Depending on the implementation, the `size()`, `length()` or `count()` methods can have a time complexity of `O(n)` where `n` is the number of elements in the collection. On the other hand, `empty()` and `is_empty()` simply check if there is at least one element in the collection, which is a constant time operation, `O(1)`.

Reviewer 2: ```suggestion
Moreover, depending on the implementation, the `size()`, `length()`, or `count()` methods can have a time complexity of `O(n)` where `n` is the number of elements in the collection. On the other hand, `empty()` and `is_empty()` simply check if there is at least one element in the collection, which is a constant time operation, `O(1)`.
```

> Using `empty()` or `is_empty()` to test for emptiness should therefore be preferred over `size()`, `length()` or `count()`.

Reviewer 2: ```suggestion
Using `empty()` or `is_empty()` to test for emptiness should therefore be preferred over `size()`, `length()`, or `count()`.
```

> == How to fix it

Reviewer 2: Regarding the content, I think the new text is an improvement.

Regarding the form, for this kind of rule where the title already explains how to fix, we try to keep the code in "Why is this an issue".
Here, instead of having the last sentence of "why is this an issue?" "Using `empty()` or `is_empty()` to test for emptiness should therefore be preferred over `size()`, `length()` or `count()`." 
I would rather have:

> Instead of checks on `size()`, `length()` or `count()`.
> [source,cpp,diff-id=1,diff-type=noncompliant]
> `----`
> ...
> `----`
> Prefer using `empty()`:
> [source,cpp,diff-id=1,diff-type=compliant]
> `----`
> ...
> `----`

== Resources

> === Documentation

Reviewer 2: ```suggestion
=== External coding guidelines
```
